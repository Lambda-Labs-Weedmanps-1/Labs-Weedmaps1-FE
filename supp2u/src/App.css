.Navigation {
  background-color: #8debac;
  height: 40px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
.Link {
  padding: 10px;
}
