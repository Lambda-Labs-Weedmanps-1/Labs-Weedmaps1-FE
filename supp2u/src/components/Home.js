import React, { useState } from 'react';
import {
	pingApiServer,
	fetchProfilesWithAuth,
	fetchProfilesNoAuth
} from '../utils/API';

import { auth } from '../utils/init';

import { Container, Row, Col, Button, ButtonGroup } from 'reactstrap';

import Loading from './Loading';

const Home = () => {
  const [status, setStatus] = useState('Idle');
  const [pingData, setPingData] = useState('No data from server yet')
  const [profileData, setProfileData] = useState('No data from server yet')

  const resetStates = () => {
    setStatus(status)
    setPingData(pingData)
    setProfileData(profileData)
  }

  const renderLoading = () => {
    switch (status) {
			case 'Fetching':
				return <Loading />;
			default:
				return null;
		}
  }

  const pingApi = () => {
    setStatus('Fetching')
		
		pingApiServer().then(data => {
      setStatus('Fetch completed')
      setPingData(data)
		});
  };
  
  const _fetchProfilesNoAuth = () => {
		setStatus('Fetching')
		fetchProfilesNoAuth().then(data => {
			setStatus('Fetch completed')
      setPingData(data)
		});
  };
  
  const _fetchProfilesWithAuth = () => {
		setStatus('Fetching')
		fetchProfilesWithAuth(this.props.token).then(data => {
			setStatus('Fetch completed')
      setPingData(data)
		});
  };
  
  const renderSignInUp = () => {
		return (
			<Button color="primary" onClick={auth.login}>
				Sign In / Sign Up
			</Button>
		);
  };
  
  const renderLogOut = () => {
		return (
			<div>
				<Row>
					<Col>
						<Button color="primary" onClick={auth.logout}>
							Sign out
						</Button>
						<Button color="secondary" onClick={pingApi}>
							Ping API Server
						</Button>
					</Col>
				</Row>
				<Row>
					<Col xs="12" sm="6">
						<Button color="danger" onClick={_fetchProfilesNoAuth}>
							Fetch Profile without Authentication
						</Button>
					</Col>
					<Col xs="12" sm="6">
						<Button color="success" onClick={_fetchProfilesWithAuth}>
							Fetch Profile with Authentication
						</Button>
					</Col>
				</Row>
				<Row>
					<Col>
						<Button outline color="danger" size="sm" onClick={resetStates}>
							Reset
						</Button>
					</Col>
				</Row>
			</div>
		);
	};


  return (
    
			<Container>
				<div className="buttons">
					<ButtonGroup vertical>
						{auth.loggedIn()
							? renderLogOut()
							: renderSignInUp()}
					</ButtonGroup>
				</div>
				{renderLoading()}

			</Container>
		);
  
}
  
	export default Home;
