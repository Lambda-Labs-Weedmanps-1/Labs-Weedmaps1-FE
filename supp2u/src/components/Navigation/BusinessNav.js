import React from 'react';
import { NavLink } from "react-router-dom";

const BusinessNav = () => {

    return (
        <div>
            <NavLink className="Link" to="/business/create">Create Business</NavLink>
        </div>
    );

}

export default BusinessNav;