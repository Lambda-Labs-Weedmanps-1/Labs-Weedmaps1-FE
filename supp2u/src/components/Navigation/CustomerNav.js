import { Link } from "react-router-dom";
import React, { useState, useCallback } from 'react'
import { useTransition, animated } from 'react-spring'

const pages = [
    ({ style }) => <animated.div style={{ ...style, background: 'lightpink' }}><Link className="Link" to="/customer/view">View Customer</Link></animated.div>,
    ({ style }) => <animated.div style={{ ...style, background: 'lightblue' }}><Link className="Link" to="/customer/view">View Customer</Link></animated.div>,
    ({ style }) => <animated.div style={{ ...style, background: 'lightpurple' }}><Link className="Link" to="/customer/view">View Customer</Link></animated.div>,
  ]


  export default function CustomerNav() {

    const [index, set] = useState(0)
    const onClick = useCallback(() => set(state => (state + 1) % 3), [])
    const transitions = useTransition(index, p => p, {
      from: { opacity: 0, transform: 'translate3d(100%,0,0)' },
      enter: { opacity: 1, transform: 'translate3d(0%,0,0)' },
      leave: { opacity: 0, transform: 'translate3d(-50%,0,0)' },
    })
    return (
      <div className="simple-trans-main" onClick={onClick}>
        {transitions.map(({ item, props, key }) => {
          const Page = pages[item]
          return <Page key={key} style={props} />
        })}
      </div>
    )
}


