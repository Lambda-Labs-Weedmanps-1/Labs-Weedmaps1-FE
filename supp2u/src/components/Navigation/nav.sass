.NavTitle
  font-size: 20px
  font-size: 3.0vw

.Navigation
  background-color: #8debac
  height: 40px
  display: flex
  flex-direction: row
  align-items: center
  justify-content: center
  font-size: 15px
  font-size: 2.5vw
  color: white

.Link
  padding: 15px