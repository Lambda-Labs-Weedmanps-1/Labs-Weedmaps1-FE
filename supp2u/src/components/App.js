import React, { Component, useState } from 'react';
// import logo from '../../assets/img/logo.svg';
import '../css/styles.styl';
import Home from './Home';
import { auth } from '../utils/init';

// * Hooks Implementation
const App = () => {
  // maybe status, isLoggedIn ???
  // const [isLoggedIn, logInUser] = useState(auth.loggedIn())
  // const [isLoggedIn, logInUser] = useState(auth.loggedIn())

  return (
    <div className="App">
      <div className="App-header">
        {/* <img src={logo} className="App-logo" alt="logo" /> */}
        <h2>Welcome to React + Auth0 + Rails API</h2>
      </div>
      <Home
        auth={auth}
        // isLoggedIn={logInUser}
        // isLoggedIn={logInUser(sisLoggedIn)}
        token={auth.getToken()}
      />
    </div>
  );
}


// class App extends Component {
// 	constructor(props) {
// 		super(props);
// 		this.state = {
// 			isLoggedIn: auth.loggedIn()
// 		};
// 	}
// 	render() {
// 		return (
// 			<div className="App">
// 				<div className="App-header">
// 					{/* <img src={logo} className="App-logo" alt="logo" /> */}
// 					<h2>Welcome to React + Auth0 + Rails API</h2>
// 				</div>
// 				<Home
// 					auth={auth}
// 					isLoggedIn={this.state.isLoggedIn}
// 					token={auth.getToken()}
// 				/>
// 			</div>
// 		);
// 	}
// }
export default App;
