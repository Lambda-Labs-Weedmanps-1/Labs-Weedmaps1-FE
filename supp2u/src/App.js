import React from 'react';
import Navigation from './components/Navigation/Navigation'
import './App.sass';
import BasicRoute from "./BasicRoute";

import { toast } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
toast.configure();


function App() {
  return (
    <div className="App">
        <Navigation />
        <BasicRoute />
    </div>
  );
}

export default App;
