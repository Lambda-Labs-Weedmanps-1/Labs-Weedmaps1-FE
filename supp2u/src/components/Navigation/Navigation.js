import React from "react";
import { Link } from "react-router-dom";
import { auth } from "../../utils/init";
import { Button } from "reactstrap";
import BusinessNav from "./BusinessNav";
import CustomerNav from "./CustomerNav";
import { withRouter } from "react-router";
import "./nav.sass";
import Image from "../../images/Image5.png"

function Navigation(props) {
  let renderLogInLogOut = () => {
    if (
      localStorage.getItem("customer_id") ||
      localStorage.getItem("business_id") ||
      localStorage.getItem("user_id")
    ) {
      return (
        <Button className="auth-button"  onClick={auth.logout}>
          Sign out
        </Button>
      );
    } else {
      return (
        <Button className="auth-button"  onClick={() => auth.login(props.history)}>
          Sign In / Sign Up
        </Button>
      );
    }
  };

  let renderUserType = () => {
    if (
      localStorage.getItem("customer_id") &&
      localStorage.getItem("user_id")
    ) {
      return <CustomerNav />;
    } else if (
      localStorage.getItem("business_id") &&
      localStorage.getItem("user_id")
    ) {
      return <BusinessNav />;
    } else {
      return;
    }
  };

  return (
    <div>
      <nav className="Navigation">
      <Link to={{ pathname: `/` }}>
        <img className="Navwidget" src={Image} alt="logo"/>
      </Link>
        {/* <img src={Image} alt="logo" width="15%" height="90%" /> */}
        {/* <Link className="Link" to="/">
          Home
        </Link> */}
        {renderUserType()}
        <div>
          {renderLogInLogOut()}
        </div>
        
      </nav>
    </div>
  );
}

export default withRouter(Navigation);
