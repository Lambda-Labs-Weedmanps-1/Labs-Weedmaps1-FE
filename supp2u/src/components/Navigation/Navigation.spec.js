import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';

import Navigation from '../Navigation/Navigation'

// Will try to text links to correct routes next
import Home from './../Home';
import App from './../../App';

// jest.mock('backend/app');

configure({ adapter: new Adapter() });

describe('Navigation Bar', () => {
    it('should render correctly', () => {
      const nabvar = shallow(<Navigation />);
    
      expect(nabvar).toMatchSnapshot();
    
    //   navbar.unmount();
    });
    
});

test('valid path should not redirect to 404', () => {
  const wrapper = mount(
    <MemoryRouter initialEntries={[ '/' ]}>
      <Navigation/>
    </MemoryRouter>
  );
  expect(wrapper.find(Navigation)).toHaveLength(1);
//   wrapper.unmount();
});

// Can't get to render correctly, will fix later
// it('render correctly text component', () => {  
//     const TextInputComponent = renderer.create(<Navigation />).toJSON();
//     expect(TextInputComponent).toMatchSnapshot();
// });