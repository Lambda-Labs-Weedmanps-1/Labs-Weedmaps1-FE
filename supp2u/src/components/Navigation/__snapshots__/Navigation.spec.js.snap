// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Navigation Bar should render correctly in "debug" mode 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
